src/domain/proposal.ts

```diff
@@ -186,12 +208,38 @@ export function updateGSProposal(
   createdAt: BigInt,
   avatarAddress: Address,
   descriptionHash: string,
+  schemeAddress: Address,
 ): void {
   let proposal = getProposal(proposalId.toHex());
   proposal.dao = avatarAddress.toHex();
   proposal.genericScheme = proposalId.toHex();
   proposal.createdAt = createdAt;
   proposal.descriptionHash = descriptionHash;
+  proposal.scheme = crypto.keccak256(concat(avatarAddress, schemeAddress)).toHex();
+  if (proposal.organizationId != null && proposal.scheme != null) {
+    setScheme(proposal.organizationId, proposal.scheme);
+    setSchemeName(proposal.scheme, 'GenericScheme');
+  }
+  // IPFS reading
+
+  let ipfsData = ipfs.cat('/ipfs/' + descriptionHash);
```

Comment:
this IPFS code is common to all schemes .
lets have it on a function .how about doing the IPFS staff of the getProposal function.

Reply:
ok